types/is-data-descriptor/index.d.ts

```diff
@@ -0,0 +1,9 @@
+// Type definitions for is-data-descriptor 1.0
+// Project: https://github.com/jonschlinkert/is-data-descriptor
+// Definitions by: Richie Bendall <https://github.com/Richienb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function isDataDesc(obj: unknown): obj is PropertyDescriptor;
+declare function isDataDesc<ObjectType>(obj: ObjectType, key: keyof ObjectType): boolean;
```

Comment:
If you don't constrain `ObjectType`, then for example the following call will be allowed, which is definitely wrong: `isDataDesc(1, 'toString')`. Strictly seen, this is allowed by the library but semantically this should be forbidden because it doesn't make sense.

You need to constrain it with `extends object`:
```suggestion
declare function isDataDesc<ObjectType extends object>(obj: ObjectType, key: keyof ObjectType): boolean;
```